src/io/hdf_dataproxy.h

```diff
@@ -47,6 +48,45 @@ inline bool get_space(hid_t grp, const std::string& aname, int rank, hsize_t* di
   return thesame;
 }
 
+/** free function to go from spec to dimensionality of memory space and data space
+ */
+template<typename IC>
+inline bool getOffsets(hid_t grp,
+		       const std::string& aname,
+		       const IC& readSpec,
+		       std::vector<hsize_t>& offset,
+		       std::vector<hsize_t>& count,
+		       int& numElements)
+{
+  std::vector<hsize_t> space_dims(readSpec.size());
+  get_space(grp, aname, readSpec.size(), space_dims.data());
+
+  // check what is requested is within bounds
+  for (int i = 0; i < readSpec.size(); i++)
+  {
+    if (readSpec[i] < -1 || readSpec[i] >= static_cast<int>(space_dims[i]))
+    {
+      APP_ABORT("Requested an invalid slice of the dataset named " + aname);
+    }
+  }
+
+  offset.resize(readSpec.size());
+  count.resize(readSpec.size());
+  numElements = 1;
+  for (int i = 0; i < readSpec.size(); i++)
+  {
+    offset[i] = readSpec[i];
+    count[i] = 1;
```

Comment:
readSpec is a mixed presentation of offsets and counts. The default count = 1 is very cryptical.